music_assistant/controllers/metadata.py

```diff
@@ -206,9 +207,7 @@ async def close(self) -> None:
     @property
     def providers(self) -> list[MetadataProvider]:
         """Return all loaded/running MetadataProviders."""
-        if TYPE_CHECKING:
-            return cast("list[MetadataProvider]", self.mass.get_providers(ProviderType.METADATA))
-        return self.mass.get_providers(ProviderType.METADATA)
+        return cast("list[MetadataProvider]", self.mass.get_providers(ProviderType.METADATA))
```

Comment:
strictly speaking, doing a cast adds a bit of runtime overhead so we try to avoid it as much as we can or hide it between a "if TYPE_CHECKING" check

Reply:
Thanks. This is new to me but I’ll take it onboard to avoid this in future.